userspace/libsinsp/filterchecks.cpp

```diff
@@ -5583,34 +5583,57 @@ int32_t sinsp_filter_check_container::parse_field_name(const char* str, bool all
 uint8_t* sinsp_filter_check_container::extract(sinsp_evt *evt, OUT uint32_t* len, bool sanitize_strings)
 {
 	*len = 0;
-	sinsp_threadinfo* tinfo = evt->get_thread_info();
-	if(tinfo == NULL)
+	string container_id;
+
+	if(evt->get_type() != PPME_CONTAINER_JSON_E)
 	{
-		return NULL;
+		sinsp_threadinfo* tinfo = evt->get_thread_info();
+		if(tinfo == NULL)
+		{
+			return NULL;
+		}
+		container_id = tinfo->m_container_id;
+	}
+	else
+	{
```

Comment:
can you add a comment here that points out that you need to parse the json from the event in the case where there is no threadinfo?